### PR TITLE
Add test for Marshal.GetHRForException

### PR DESCRIPTION
**Pull request by jkotas**
Includes regression test for #11144

### CODE REVIEW

**Comment by jkotas** [comment]
Won't pass until https://github.com/dotnet/coreclr/pull/6929 arrives to corefx.

**Comment by jkotas** [comment]
@yizhang82 PTLA

**Comment by yizhang82** [comment]
LGTM. Thanks!

File: src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.cs [hunk 1]
```diff
@@ -128,5 +128,15 @@ private static void VerifyBytes(IntPtr p, int length)
...
             }
         }
+
+        [Fact]
+        public static void GetHRForException()
+        {
+            Assert.Equal(0, Marshal.GetHRForException(null));
+
+            Exception e = new Exception();
+            Assert.True(Marshal.GetHRForException(e) < 0);
```
Comment (stephentoub) [inline src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.cs]: Nit: Could be:
```
Assert.InRange(Marshal.GetHRForException(e), int.MinValue, -1);
```
The small advantage there is that the error message that'll be generated if this fails will include the details on the actual value and the range.

**Comment by stephentoub** [comment]
LGTM

File: src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.cs (hunk 1 above)
Comment (jkotas) [inline src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.cs]: Thanks - fixed.

**Comment by jkotas** [comment]
@dotnet-bot test this please

**Comment by jkotas** [comment]
@dotnet-bot test Innerloop OSX Debug Build and Test  please

**Pull request merged**